converter/input/mgologstash/flow.go

```diff
@@ -412,6 +412,137 @@ func (i *Flow) fillFromNetflowv9BSONMap(netflowMap bson.M) error {
+		sourceIP, ok = sourceIPIface.(string)
+		if !ok {
+			return errors.Errorf("could not convert %+v to string", sourceIPIface)
+		}
+	} else {
+		return errors.New("input map must contain key 'netflow.ipv4_src_addr'")
+	}
+
+	sourcePortIface, ok := netflowMap["l4_src_port"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.l4_src_port'")
+	}
+	sourcePort, ok := sourcePortIface.(int)
+	if !ok {
+		return errors.Errorf("could not convert %+v to int", sourcePortIface)
+	}
+
+	var destIP string
+	destIPIface, destIPOk := netflowMap["ipv4_dst_addr"]
+	if destIPOk {
+		destIP, ok = destIPIface.(string)
+		if !ok {
+			return errors.Errorf("could not convert %+v to string", destIPIface)
+		}
+	} else {
+		return errors.New("input map must contain key 'netflow.ipv4_dst_addr'")
+	}
+
+	var destPort int
+	destPortIface, ok := netflowMap["l4_dst_port"]
+	if ok {
+		destPort, ok = destPortIface.(int)
+		if !ok {
+			return errors.Errorf("could not convert %+v to int", destPortIface)
+		}
+	} else {
+		return errors.New("input map must contain key 'netflow.l4_dst_port'")
+	}
+
+	flowStartIface, ok := netflowMap["first_switched"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.first_switched'")
+	}
+	flowStart, ok := flowStartIface.(string)
+	if !ok {
+		return errors.Errorf("could not convert %+v to string", flowStartIface)
+	}
+
+	flowEndIface, ok := netflowMap["last_switched"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.last_switched'")
+	}
+	flowEnd, ok := flowEndIface.(string)
+	if !ok {
+		return errors.Errorf("could not convert %+v to string", flowEndIface)
+	}
+
+	octetTotalIface, ok := netflowMap["in_bytes"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.in_bytes'")
+	}
+	octetTotal, ok := octetTotalIface.(int64)
+	if !ok {
+		//Logstash creates these fields as 32 bit ints,
+		//Go handles them as 64 bit ints, provide both casts
+		octetTotal32, octetTotal32Ok := octetTotalIface.(int)
+		if !octetTotal32Ok {
+			return errors.Errorf("could not convert %+v to int", octetTotalIface)
+		}
+		octetTotal = int64(octetTotal32)
+	}
+
+	packetTotalIface, ok := netflowMap["in_pkts"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.in_pkts'")
+	}
+	packetTotal, ok := packetTotalIface.(int64)
+	if !ok {
+		//Logstash creates these fields as 32 bit ints,
+		//Go handles them as 64 bit ints, provide both casts
+		packetTotal32, packetTotal32Ok := packetTotalIface.(int)
+		if !packetTotal32Ok {
+			return errors.Errorf("could not convert %+v to int", packetTotalIface)
+		}
+		packetTotal = int64(packetTotal32)
+	}
+
+	protocolIDIface, ok := netflowMap["protocol"]
+	if !ok {
+		return errors.New("input map must contain key 'netflow.protocol'")
+	}
+	protocolID, ok := protocolIDIface.(int)
+	if !ok {
+		return errors.Errorf("could not convert %+v to int", protocolIDIface)
+	}
+
+	//Fill in the flow now that we know we have all the data
+	if sourceIPOk {
```

Comment:
We can remove the if checks down here for sourceIPOk and destIPOk since we only check IPv4 and we do the checks above.

If you want to leave them, it shouldn't hurt anything.